azure-webapp-maven-plugin/README.md

```diff
@@ -1,186 +1,177 @@
+                            <resource>
+                                <directory>${project.basedir}/target</directory>
+                                <includes>
+                                    <include>*.war</include>
+                                </includes>
+                            </resource>
+                        </resources>
+                    </deployment>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>      
     ```
-   
-3. Use the following commands to deploy your project to Azure App Service. 
-
+3. Use the following commands to deploy your project to Azure App Service.
     ```
     $ mvn azure-webapp:deploy
     ```
 
-<a name="samples"></a>
-## Samples	
-A few typical usages of Maven Plugin for Azure App Service Web Apps are listed at [Web App Samples](../docs/web-app-samples.md). You can choose one to quickly get started.
-
 <a name="goals"></a>
 ## Goals
 
-The Maven Plugin for Azure App Service Web Apps has only one goal: `azure-webapp:deploy`. 
+#### `azure-webapp:config`
+-  A plugin goal providing a great experience for managing the configuration of [azure-webapp:deploy](#azure-webappdeploy), it will rewrite your pom configuration based on your input. Details about the configuration could be found in the following.
+-  The plugin goal will divide the configuration into 3 parts.
+   1. Application
+        
+        You can configure appName, resource group, region, pricing tier at this part.
+   2. Runtime
+        
+        You can configure operating system and runtime stack at this part. It should be noted that you can not change the operating system after you deploy it.
+   3. DeploymentSlot
+        
+        You can configure deployment slot at this part.
+
+#### `azure-webapp:deploy`
+- Deploy artifacts or docker container image to an Azure Web App based on your configuration.<br>If the specified Web App does not exist, it will be created.
+- Configuration:
+    Common configurations of all Maven Plugins for Azure can be found [here](../docs/common-configuration.md).
+
+    This maven plugin supports two kinds of configurations V2 and V1 (deprecated). Specify the configuration
+    `<schemaVersion>V2</schemaVersion>` to use the V2 configuration. Strongly suggest use v2 schema.
+    You can find v1 schema [here](v1-schema.md)
+
+    Property | Required | Description | Version
+    ---|---|---|---
+    [region](#region) | true | Specifies the region where your Web App will be hosted; the default value is **westus**. All valid regions at [Supported Regions](#region) section. | 0.1.0+
+    `<resourceGroup>` | true | Azure Resource Group for your Web App. | 0.1.0+
+    `<appName>` | true | The name of your Web App. | 0.1.0+
+    [pricingTier](#pricingtier) | false | The pricing tier for your Web App. The default value is **P1V2**.| 0.1.0+
+    `<deploymentSlot>` | false | The deployment slot to deploy your application. | 1.3.0+
```

Comment:
Do we have more explanation about the `deploymentSlot` and the `appSettings`?

Reply:
Has already added some app settings example in samples. 
We can also provide a link to guide users find deployment slot examples in samples

Reply:
Sounds good for adding links